projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,201 @@
+#include "ads1015.h"
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+static void prv_check_status(StatusCode code) {
```

Comment:
This won't work - look at the implementation of `status_msg`. It returns a StatusCode that also logs the current line, so any error you get will be completely useless. Also, this is a void-type return, so you can't return StatusCodes.

You should have tests that validate out-of-bounds and invalid conditions so you would've been able to see this fail.

Reply:
I will modify this. 
Am I returning a StatusCode? because I do not have a return line.

Reply:
No, this will not work at all. Your line would be wrong and you have no return type.